nano/nano_node/daemon.cpp

```diff
@@ -128,6 +128,14 @@ void nano_daemon::daemon::run (boost::filesystem::path const & data_path, nano::
 				logger.always_log (boost::format ("Open file descriptors limit is %1%") % file_descriptor_limit);
 			}
 
+			// a 0-valued port means for the node 'let the OS decide'; however, for the daemon start up, if
+			// the user hasn't specified a port in the config, we must use the default peering port for the network
+			//
+			if (!config.node.peering_port)
+			{
+				config.node.peering_port = network_params.network.default_node_port;
+			}
+
```

Comment:
This looks good to me. Personally, I would have the following for maximum flexibility:
* convert peering_port from uint64_t to int.
* Set the default value to "-1" to specify the value is not set
* if the value is -1 then do `config.node.peering_port = network_params.network.default_node_port`

That would allow for 0 to mean, select the port number yourself and be consistent.
But I am happy with the change as is, as well.

Reply:
It is true that currently `0` means "there hasn't been any explicit choice made" (which is fine, no magic meaning), and which further might mean "let the OS decide" (which is again fine, it's an idiomatic meaning for the port value), or, could also mean "we have to set the default value" if it's the daemon who asks the question. I am not extremely happy about the last part, but it's a very reasonable magic meaning IMO.

I'd rather keep the port an unsigned as it's more semantic for a port value; having yet another magic number (`-1` in this case) to only represent that last case brings some temporary joy but leaves room for problems down the way IMHO. But yeah, very decent thoughts, thanks for the review.

Reply:
std::optional would be the canonical way to indicate whether a value was specified or not. Then if it's std::nullopt_t we know it wasn't specified and if it's 0 we know it was specified to 0.